proto/connector/v2/connector.proto

```diff
@@ -0,0 +1,316 @@
+  rpc Stop(Destination.Stop.Request) returns (Destination.Stop.Response);
+  // Teardown signals to the plugin that there will be no more calls to any
+  // other function. After Teardown returns, the plugin should be ready for a
+  // graceful shutdown.
+  rpc Teardown(Destination.Teardown.Request) returns (Destination.Teardown.Response);
+
+  // -- Lifecycle events -------------------------------------------------------
+
+  // LifecycleOnCreated is called after Configure and before Start when the
+  // connector is run for the first time. This call will be skipped if a
+  // connector was already started before. This method can be used to do some
+  // initialization that needs to happen only once in the lifetime of a
+  // connector (e.g. create a bucket). Anything that the connector creates in
+  // this method is considered to be owned by this connector and should be
+  // cleaned up in LifecycleOnDeleted.
+  rpc LifecycleOnCreated(Destination.Lifecycle.OnCreated.Request) returns (Destination.Lifecycle.OnCreated.Response);
+  // LifecycleOnUpdated is called after Configure and before Start when the
+  // connector configuration has changed since the last run. This call will be
+  // skipped if the connector configuration did not change. It can be used to
+  // update anything that was initialized in LifecycleOnCreated, in case the
+  // configuration change affects it.
+  rpc LifecycleOnUpdated(Destination.Lifecycle.OnUpdated.Request) returns (Destination.Lifecycle.OnUpdated.Response);
+  // LifecycleOnDeleted is called when the connector was deleted. It will be the
+  // only method that is called in that case. This method can be used to clean
+  // up anything that was initialized in LifecycleOnCreated.
+  rpc LifecycleOnDeleted(Destination.Lifecycle.OnDeleted.Request) returns (Destination.Lifecycle.OnDeleted.Response);
+}
+
+// SpecifierPlugin is responsible for returning the plugin specification.
+service SpecifierPlugin {
+  // Specify should return the plugin specification.
+  rpc Specify(Specifier.Specify.Request) returns (Specifier.Specify.Response);
+}
+
+message Source {
+  message Configure {
+    message Request {
+      // Config contains the raw plugin settings.
+      map<string, string> config = 1;
+    }
+    message Response {}
+  }
+
+  message Start {
+    message Request {
+      // This is the position of the last record that was successfully
+      // processed. The Source should start producing records after this
+      // position.
+      bytes position = 1;
+    }
+    message Response {}
+  }
+
+  message Run {
+    message Request {
+      // These are the positions of the records that were successfully
+      // processed.
+      repeated bytes ack_positions = 1;
+    }
+    message Response {
+      // Records contains the OpenCDC records read by the source from the 3rd
+      // party resource.
+      repeated opencdc.v1.Record records = 1;
+    }
```

Comment:
Notice that in the v2 protocol the source can send a batch of records to Conduit, while Conduit can send a batch of acknowledgments back.